CONTRIBUTING.md

```diff
@@ -1,44 +1,199 @@
+
+Enhancements are tracked as [GitHub Issues](https://guides.github.com/features/issues/). To submit new enhancement, create new Issue and incllude these details:
+
+- **Title**
+    - Write prefix `[Enhancement]`
+    - Use a clear and descriptive title
+- **Body** - include the following sections:
+    - *Motivation* - why do we need it?
+    - *Target* - what is going to be improved?
+    - *Description* - how to implement it?
+
+
+
+### Asking Questions
+
+A *question* is any discussion that is typically neigher a bug, nor feature requests, nor improvements - "How do I do X?".
+
+Questions are tracked as [Github Issues](https://guides.github.com/features/issues/) or via private messages in [your favourite messenger](#contact-developers).
+
+To submit new question in GitHub Issues, it must include these details:
+
+- **Title**
+    - Write prefix `[Question]`
+    - Use a clear and descriptive title
+- **Body** - describe your question with as many details as possible.
+
+
+
+### Your First Code Contribution
+
+Read our [C++ Style Guide](#c++-style-guide) and start with beginner-friendly issues with label [`[good-first-issue]`](https://github.com/hyperledger/iroha/issues?q=is:open+is:issue+label:good-first-issue ). 
+
+
+
+### Pull Requests
+
+- Fill in [the required template](.github/PULL_REQUEST_TEMPLATE.md)
+
+- **Write tests** for new code.
+
+- Every pull request should be reviewed and **get at least two approvals**.
+
+- Do not include issue numbers in the PR title or commit messages.
+
+- Include issue numbers in Pull Request body only.
+
+- When finished work, **rebase onto base branch** with 
+    ```bash
+    $ git fetch
+    $ git rebase -i <base-branch>
+    ```
+
+    [Step-by-step guide](https://soramitsu.atlassian.net/wiki/spaces/IS/pages/11173889/Rebase+and+merge+guide).
+
+- Follow the [C++ Style Guide](#C++-style-guide).
+
+- Follow the [Git Style Guide](#git-commit-messages) .
+
+- **Document new code** based on the [Documentation Styleguide](#documentation-styleguide)
+
+- End all files with a newline.
+
+
+
+
+## Styleguides
+
+### Git Style Guide
+
+- **Use present tense** ("Add feautre", not "Added feature").
+- **Use imperative mood** ("Deploy docker to..." not "Deploys docker to...").
+- Write meaningful commit message.
+- **Sign every commit** with [DCO](https://github.com/apps/dco): `Signed-off-by: $NAME <$EMAIL>`. 
+    You can do it automatically using `git commit -s`.
+- Do not include PR or Issue number in commit message. 
+- Limit the first line of commit message to 50 characters or less.
+- When only changing documentation, include `[ci skip]` in the commit description.
+- We use mixed approach of [Github Flow](https://guides.github.com/introduction/flow/) and [Git Flow](http://nvie.com/posts/a-successful-git-branching-model/). More at [Iroha Working Agreement](https://github.com/hyperledger/iroha/wiki/Iroha-working-agreement#2-version-control-system).
+
```

Comment:
Adding a blank line after first one in the commit message is also a good practice